lib/models/obms.js

```diff
@@ -0,0 +1,183 @@
+// Copyrighr 2016, EMC, Inc.
+
+'use strict';
+
+module.exports = ObmsModelFactory;
+
+ObmsModelFactory.$provide = 'Models.Obms';
+ObmsModelFactory.$inject = [
+    'Model',
+    'Services.Waterline',
+    '_',
+    'Services.Encryption',
+    'Constants',
+    'Promise',
+    'util',
+    'Errors',
+    'Assert'
+];
+
+function ObmsModelFactory (
+    Model,
+    waterline,
+    _,
+    encryption,
+    Constants,
+    Promise,
+    util,
+    Errors,
+    assert
+) {
+    var secretsPattern = _.reduce(Constants.Logging.Redactions, function(pattern, regex) {
```

Comment:
The `secret` is a good idea, but it can only cover the case that we have solid key in model.  Take the Task options for example, the task model only defines a key `options`, but the properties in `options` may still contain some sensitive data, like the rootPassword in install-os task. You may think that we can extend this key in task schema as well. But seems the task model is hard to access the external task schema.